File: backend_Myecoprint/backend_myecoprint/models/contcat.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Store= require('./store');

const Contact = sequelize.define('Contact ', {
    contact_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
    },
    store_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },

    phone_number: {
        type: DataTypes.STRING,
    },
    branch: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    createdAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
    updatedAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
    deletedAt: {
        type: DataTypes.DATE,
        allowNull: true,
    },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'contcats', 
  });
  Contact.belongsTo(Store, { foreignKey: 'store_id', as: 'stores',onDelete: 'CASCADE' });
module.exports =Contact;

File: Myecoprint_admin/ncej_admin/lib/data/module/contcat.dart

class Contact {
  int contactId;
  int storeId;
  String? phoneNumber;
  String branch;


  Contact({
    required this.contactId,
    required this.storeId,
    this.phoneNumber,
    required this.branch,
 
  });

  factory Contact.fromJson(Map<String, dynamic> json) {
    return Contact(
      contactId: json['contact_id']as int? ??0,
      storeId: json['store_id']as int? ??0,
      phoneNumber: json['phone_number']as String? ?? '',
      branch: json['branch'],
 
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'contact_id': contactId,
      'store_id': storeId,
      'phone_number': phoneNumber,
      'branch': branch,
 
    };
  }
}
